cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -110,6 +110,11 @@ func SetNodeRegistrationDynamicDefaults(cfg *kubeadmapi.NodeRegistrationOptions,
 	}
 
 	if cfg.CRISocket == "" {
+		if skipCRIDetect {
+			klog.V(4).Infof("skip CRI socket detection, fill with placeholder %s", kubeadmconstants.UnknownCRISocket)
+			cfg.CRISocket = kubeadmconstants.UnknownCRISocket // set a value to pass the ValidateSocketPath
+			return nil
+		}
```

Comment:
Do we have any testcase to check the logic added here?

Reply:
This can only happen if the execution of the command or sub-command does not depend on the `CRISocket`. The purpose here is just to pass the validation check. It's pretty straightforward. 
Previously, we were pre-setting the CRISocket field to prevent CRI detection, for example:
https://github.com/kubernetes/kubernetes/blob/fc479f41b3d2130b18b83aa12b8cecb83d67a9c1/cmd/kubeadm/app/cmd/util/cmdutil.go#L108-L115
https://github.com/kubernetes/kubernetes/blob/fc479f41b3d2130b18b83aa12b8cecb83d67a9c1/cmd/kubeadm/app/cmd/token_test.go#L176-L178
Currently, we use this additional parameter `skipCRIDetect`  to control whether to do CRI detection. This seems easier to understand than implicit control.

BTW, I think we need to do a subsequent refactoring with the struct option, and that might be a good time to add more test cases.